spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.dispatcher;
+
+import static org.opensearch.sql.spark.data.constants.SparkConstants.FLINT_DEFAULT_AUTH;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.FLINT_DEFAULT_HOST;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.FLINT_DEFAULT_PORT;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.FLINT_DEFAULT_REGION;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.FLINT_DEFAULT_SCHEME;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.FLINT_INTEGRATION_JAR;
+
+import lombok.AllArgsConstructor;
+import org.opensearch.sql.datasource.DataSourceService;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.spark.client.EmrServerlessClient;
+import org.opensearch.sql.spark.helper.FlintHelper;
+
+@AllArgsConstructor
+public class SparkQueryDispatcher {
+
+  private EmrServerlessClient emrServerlessClient;
+
+  private DataSourceService dataSourceService;
+
+  public String dispatch(String applicationId, String query) {
+    String datasourceName = getDataSourceName(query);
+    return emrServerlessClient.startJobRun(
+        applicationId, query, getExecutionRoleARN(datasourceName), getFlintHelper(datasourceName));
+  }
+
+  private String getDataSourceName(String query) {
```

Comment:
can this be a separate datasource for glue or more generalized a catalog based datasource ?